DOC Add datasets section to website (#720)
skrub-data/skrub: doc/datasets.rst

.. _datasets:

> ========
> Datasets
> ========

Reviewer: ```suggestion
================
Example datasets
================
```

.. currentmodule:: skrub

skrub provides fetching methods for datasets, 
which are generally used in examples and benchmarks.

It consists of generated, embeddings and real world data.

> Real world datasets

Reviewer: It looks to me like we are duplicating the API page. I would rather that we point to it rather than establish a second list, as this list will be hard to keep up to date.

Reviewer: But why do that? It adds complexity to the API page.

I would rather we don't change the API page, and just do a very small page on the datasets in the narrative docs that points to the API page. The datasets are not a core user-facing aspect of skrub. They are there to enable us to do good examples. They don't need to be prominent in the doc

Author: The idea was to list all functions in both the API and datasets file while not duplicating them explicitly.

I understand the complexity it adds and that duplicating is not the best option. 
Here is a very simple alternative version where I only provide a link to the API page.

.. _datasets:

> =========
> Datasets
> =========

Reviewer 2: ```suggestion
========
Datasets
========
```

.. currentmodule:: skrub

> skrub allows you to fetch different datasets used in
> examples and benchmarks.
> 
> It consists of various methods to get generated, embeddings and `real world` data.

Reviewer 2: ```suggestion
skrub provides fetching methods for datasets, 
which are generally used in examples and benchmarks.

It consists of generated, embeddings and real world data.
```

Real world datasets
-------------------

> From openml.org:
> 
> * :func:`datasets.fetch_employee_salaries`
> * :func:`datasets.fetch_road_safety`
> * :func:`datasets.fetch_medical_charge`
> * :func:`datasets.fetch_midwest_survey`
> * :func:`datasets.fetch_open_payments`
> * :func:`datasets.fetch_traffic_violations`
> * :func:`datasets.fetch_drug_directory`
> 
> From other sources:
> 
> * :func:`datasets.fetch_world_bank_indicator`
> * :func:`datasets.fetch_figshare`

Reviewer 2: I don't think it's relevant for the user to know where they come from.
```suggestion
* :func:`datasets.fetch_employee_salaries`
* :func:`datasets.fetch_road_safety`
* :func:`datasets.fetch_medical_charge`
* :func:`datasets.fetch_midwest_survey`
* :func:`datasets.fetch_open_payments`
* :func:`datasets.fetch_traffic_violations`
* :func:`datasets.fetch_drug_directory`
* :func:`datasets.fetch_world_bank_indicator`
* :func:`datasets.fetch_figshare`
```
